File: script/get_employees.js
const axios = require('axios');
const { MongoClient } = require('mongodb');
const fs = require('fs');

async function fetchEmployees() {
  try {
    const response = await axios.get('https://soul-connection.fr/api/employees', {
      headers: {
        'accept': 'application/json',
        'X-Group-Authorization': 'bf0d22904b98ad48a9cbf9251758ce74',
        'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MywiZW1haWwiOiJqZWFubmUubWFydGluQHNvdWwtY29ubmVjdGlvbi5mciIsIm5hbWUiOiJKZWFubmUiLCJzdXJuYW1lIjoiTWFydGluIiwiZXhwIjoxNzI3MTY2MTQwfQ.HugTXnTMkWuesiXUujLoTiwdJ3VVqqKDGwv2dk76oFo'
      }
    });
    return response.data;
  } catch (error) {
    console.error('Error fetching employees:', error);
    return null;
  }
}

async function storeEmployeesInMongoDB(employees) {
  const uri = 'mongodb://localhost:27017';
  const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });

  try {
    await client.connect();
    const database = client.db('soul-connection');
    const collection = database.collection('employees');

    const result = await collection.insertMany(employees);
    console.log(`${result.insertedCount} employees were inserted into the collection.`);
  } catch (error) {
    console.error('Error storing employees in MongoDB:', error);
  } finally {
    await client.close();
  }
}

async function storeEmployeesInJsonFile(employees) {
    const filePath = './employees.json';
    try {
      fs.writeFileSync(filePath, JSON.stringify(employees, null, 2));
      console.log(`Employees data was written to ${filePath}`);
    } catch (error) {
      console.error('Error writing employees to JSON file:', error);
    }
  }

(async function () {
  const employees = await fetchEmployees();
  if (employees) {
    await storeEmployeesInMongoDB(employees);
    storeEmployeesInJsonFile(employees);
  }
})();

File: script/get_employees.py
#!/usr/bin/env python

import requests
from pymongo import MongoClient

if __name__ == "__main__":
    url = "https://soul-connection.fr/api/employees"
    api_key = "bf0d22904b98ad48a9cbf9251758ce74"
    jwt_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTksImVtYWlsIjoiZXRpZW5uZS5yb3Vzc2VsQHNvdWwtY29ubmVjdGlvbi5mciIsIm5hbWUiOiJFdGllbm5lIiwic3VybmFtZSI6IlJvdXNzZWwiLCJleHAiOjE3MjcyNTMyNjh9.xljCIaLaGdI7d9OeSlYJkRFkiDE_4glqXagdCdGbPL8"

    client = MongoClient("mongodb://localhost:27017/")
    db = client.soul_connection
    employeescollection = db.employees

    headers = {
        "X-Group-Authorization": api_key,
        "Authorization": f"Bearer {jwt_token}"
    }

    datas = []

    response = requests.get(url, headers=headers)

    employees = response.json()

    for employee in employees:
        urlemployees = "https://soul-connection.fr/api/employees/" + str(employee['id'])
        responseemployees = requests.get(urlemployees, headers=headers)
        employee = responseemployees.json()

        data = {'employee_id' : employee['id'],
                'email' : employee['email'],
                'name' : employee['name'],
                'surname' : employee['surname'],
                'birth_date' : employee['birth_date'],
                'gender' : employee['gender'],
                'work' : employee['work'],
                'image' :   0
                }

        try:
            urlclothes = "https://soul-connection.fr/api/employees/" + str(employee['id']) + "/image"
            responseimage = requests.get(urlclothes, headers=headers)
            image = responseimage.content
        except:
            image = 0

        data['image'] = image

        #datas.append(data)
        print (data['employee_id'])
        employeescollection.insert_one(data)
    

    print("fin")